relay-server/src/actors/project.rs

```diff
@@ -859,8 +863,8 @@ impl Project {
     /// outdated.
     pub fn enqueue_sampling(&mut self, message: ProcessEnvelope) {
         match self.get_cached_state(message.envelope.meta().no_cache()) {
-            Some(_) => self.flush_sampling(message),
-            None => self.pending_sampling.push_back(message),
+            Some(state) if !state.invalid() => self.flush_sampling(message),
```

Comment:
With our reliance on dynamic sampling this makes sense, but then please also change the signature:
- Pass `state` as parameter to `flush_sampling`
- Remove the "Ignore all errors" comment and state access within.